Cargo.toml

```diff
@@ -46,6 +46,14 @@ path = "sdk/typespec/typespec_macros"
 [workspace.dependencies.azure_core]
 version = "0.23.0"
 path = "sdk/core/azure_core"
+default-features = false
```

Comment:
Is this going to be problematic for new packages? Won't we need to be caeful to ensure the right features are enabled?

Reply:
The line below pulls all the same stuff in, hence my comment. This came up because I was getting a warning that @LarryOsterman's stuff had `default-features = false` and would be ignored. The end result of this change should be net zero.